paasta_tools/paasta_api.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Responds to paasta service and instance status requests.
+"""
+import argparse
+import logging
+import traceback
+
+import flask
+import requests_cache
+from werkzeug.exceptions import default_exceptions
+from werkzeug.exceptions import HTTPException
+
+from paasta_tools import marathon_tools
+from paasta_tools.cli.cmds.status import get_actual_deployments
+from paasta_tools.marathon_serviceinit import get_bouncing_status
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import validate_service_instance
+
+
+log = logging.getLogger(__name__)
+app = flask.Flask("paasta_api")
+
+
+# WORK IN PROGRESS
+def chronos_service_status(instance_status,
+                           service, instance, cluster, soa_dir, verbose):
+    return
+
+
+# INCOMPLETE
+def marathon_service_status(instance_status,
+                            service, instance, cluster, soa_dir, verbose):
+
+    marathon_config = marathon_tools.load_marathon_config()
+    client = marathon_tools.get_marathon_client(marathon_config.get_url(),
+                                                marathon_config.get_username(),
+                                                marathon_config.get_password())
+    job_config = marathon_tools.load_marathon_service_config(
+        service, instance, cluster, soa_dir)
+
+    instance_status['State'] = PaastaColors.decolor(
+        get_bouncing_status(service, instance, client, job_config))
+    instance_status['Desired state'] = PaastaColors.decolor(
+        job_config.get_desired_state_human())
+
+
+@app.route('/paasta_api/v1/status/<string:service>', methods=['GET'])
```

Comment:
Indeed. Check out the description on blueprints here http://flask.pocoo.org/docs/0.11/blueprints/#blueprints